Mathlib/Algebra/Star/LinearMap.lean

```diff
@@ -119,3 +121,47 @@ theorem intrinsicStar_rTensor (f : E →ₗ[R] F) : star (rTensor G f) = rTensor
 end TensorProduct
 
 end LinearMap
+
+namespace Module.End
+
+open scoped IntrinsicStar
+
+/-- Intrinsic star operation for `(End R E)ˣ`. -/
+def Units.intrinsicStar : Star (End R E)ˣ where
+  star f := by
+    refine ⟨star f, star (f⁻¹ : (End R E)ˣ), ?_, ?_⟩
+    all_goals
+      rw [mul_eq_comp, ← LinearMap.intrinsicStar_comp]
+      simp [← mul_eq_comp, one_eq_id]
+
+scoped[IntrinsicStar] attribute [instance] Module.End.Units.intrinsicStar
+
+theorem IsUnit.intrinsicStar {f : End R E} (hf : IsUnit f) :
+    IsUnit (star f) :=
+  have ⟨u, hu⟩ := hf
+  hu ▸ (star u).isUnit
+
+open Module.End in
+@[simp] theorem isUnit_intrinsicStar_iff {f : End R E} :
+    IsUnit (star f) ↔ IsUnit f :=
+  ⟨fun h ↦ star_star f ▸ h.intrinsicStar, fun h ↦ h.intrinsicStar⟩
+
+section eigenspace
+variable {R V : Type*} [InvolutiveStar R] [AddCommGroup V] [StarAddMonoid V]
+
```

Comment:
I think we should just add:
```suggestion
  [CommRing R] [Module R V] [StarModule R V]

```
here and not bother with the fact that `spectrum_intrinsicStar` can be proved with `CommSemiring`.